### PR TITLE
Added pre-commit config and fixes

### PR DESCRIPTION
**Pull request by amolkahat**
This patch adds pre-commit config and
fixes related to it.

Signed-off-by: Amol Kahat <amolkahat@gmail.com>

### CODE REVIEW

**Comment by yazug** [comment]
lgtm, looks like standard whitespace cleanup besides the precommit hook.

**Review by fmount — APPROVED** [review]

File: config/crd/bases/designate.openstack.org_designateapis.yaml [hunk 1]
```diff
@@ -129,6 +129,26 @@ spec:
                 description: Resources - Compute Resources required by this service
                   (Limits/Requests). https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
                 properties:
+                  claims:
```
Comment (marios) [inline config/crd/bases/designate.openstack.org_designateapis.yaml:132]: @amolkahat what are these changes this seems to go beyond just fixes for precommit?

is this related to precommit?

File: Makefile [hunk 2]
```diff
@@ -102,6 +102,15 @@ vet: gowork ## Run go vet against code.
...
 
+.PHONY: tidy
+tidy: fmt
+	go mod tidy
+
+.PHONY: golangci-lint
+golangci-lint:
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.51.2
+	$(LOCALBIN)/golangci-lint run --fix
+
```
Comment (rlandy) [inline Makefile:113]: Addition needed to run job.

File: config/rbac/service_account.yaml [hunk 3]
```diff
@@ -53,4 +53,3 @@ subjects:
 - kind: ServiceAccount
   name: designate
   namespace: openstack
```
Comment (rlandy) [inline config/rbac/service_account.yaml:55]: do we know which is the right one to delete?

**Review by rlandy — COMMENTED** [review]
Please add a local test run results

**Comment by amolkahat** [comment]
Pre-commit run output
```
akahat@jatayu designate-operator $ pre-commit run -a
golangci-lint............................................................Passed
gofmt....................................................................Passed
govet....................................................................Passed
gotidy...................................................................Passed
make-manifests...........................................................Passed
make-generate............................................................Passed
make-operator-lint.......................................................Passed
check for added large files..............................................Passed
fix utf-8 byte order marker..............................................Passed
check for case conflicts.................................................Passed
check that executables have shebangs.....................................Passed
check that scripts with shebangs are executable..........................Passed
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
detect destroyed symlinks................................................Passed
check yaml...............................................................Passed
check json...............................................................Passed
detect private key.......................................................Passed
fix end of files.........................................................Passed
don't commit to branch...................................................Passed
trim trailing whitespace.................................................Passed
bashate..................................................................Passed

akahat@jatayu designate-operator $
```

File: config/rbac/service_account.yaml (hunk 3 above)
Comment (amolkahat) [reply config/rbac/service_account.yaml:55]: may be @dkehn can help us here.

File: config/crd/bases/designate.openstack.org_designateapis.yaml (hunk 1 above)
Comment (amolkahat) [reply config/crd/bases/designate.openstack.org_designateapis.yaml:132]: This changes are added by `operator-lint `.

Comment (marios) [reply config/crd/bases/designate.openstack.org_designateapis.yaml:132]: ok

i did some digging I think this is related to Persistent Volume Claims? I found that https://qubitpi.github.io/kubernetes-website/docs/reference/kubernetes-api/config-and-storage-resources/persistent-volume-claim-v1/#PersistentVolumeClaimSpec

where it talks about resources.claims as well as the .limits and .requests that follow it

File: .github/create_bundle.sh [hunk 4]
```diff
@@ -34,32 +34,32 @@ cat "${CLUSTER_BUNDLE_FILE}" | grep "image:"
...
     else
-      echo "$base_image${delimeter}$tag_image not changed"
+        digest_image=$(skopeo inspect docker://${base_image}${delimeter}${tag_image} | jq '.Digest' -r)
+        echo "Base image: $base_image"
+        if [ -n "$digest_image" ]; then
+            echo "$base_image${delimeter}$tag_image becomes $base_image@$digest_image"
+            sed -i "s|$base_image$delimeter$tag_image|$base_image@$digest_image|g" "${CLUSTER_BUNDLE_FILE}"
+        else
+            echo "$base_image${delimeter}$tag_image not changed"
+        fi
```
[the remark this answers is not in the document]
Comment (amolkahat) [reply .github/create_bundle.sh:61]: Yeah.. at first look it feels like logic is changed, but only spaces are added. No change in logic.

**Review by marios — APPROVED** [review]
looks ok though I'd still be happier to understand the addition of claims by the linter

**Review by dkehn — APPROVED** [review]
LGTM
